pyClient/zethSimpleBenchmark.py

```diff
@@ -0,0 +1,265 @@
+                ]
+
+                output_note1 = zethGRPC.createZethNote(zethGRPC.noteRandomness(), bob_apk, zethGRPC.int64ToHexadecimal(Web3.toWei('2', 'ether')))
+                output_note2 = zethGRPC.createZethNote(zethGRPC.noteRandomness(), bob_apk, zethGRPC.int64ToHexadecimal(Web3.toWei('2', 'ether')))
+                js_outputs = [
+                    output_note1,
+                    output_note2
+                ]
+
+                proof_input = zethGRPC.makeProofInputs(initial_root, js_inputs, js_outputs, zethGRPC.int64ToHexadecimal(Web3.toWei('4', 'ether')), zero_wei_hex)
+
+                proof_time_start = time.time()
+                proof_obj = zethGRPC.getProof(test_grpc_endpoint, proof_input)
+                proot_time_end = time.time()
+                proof_duration = proot_time_end - proof_time_start
+
+                proof_json = zethGRPC.parseProof(proof_obj, zksnark)
+
+                output_note1_str = json.dumps(zethGRPC.parseZethNote(output_note1))
+                output_note2_str = json.dumps(zethGRPC.parseZethNote(output_note2))
+
+                ciphertext1 = zethUtils.encrypt(output_note1_str, keystore["Bob"]["AddrPk"]["ek"])
+                ciphertext2 = zethUtils.encrypt(output_note2_str, keystore["Bob"]["AddrPk"]["ek"])
+
+                result_deposit_bob_to_bob, gas_cost, mix_duration, trans_duration =  mix(
+                    mixer_instance,
+                    ciphertext1,
+                    ciphertext2,
+                    proof_json,
+                    bob_eth_address,
+                    w3.toWei(4, 'ether'),
+                    4000000,
+                    zksnark
+                )
+                test_time_end = time.time()
+                tot_duration = proof_duration+ mix_duration+ trans_duration
+
+                proof_dts = np.append(proof_dts, [proof_duration])
+                mix_dts = np.append(mix_dts, [mix_duration])
+                tx_dts = np.append(tx_dts, [trans_duration])
+                tot_dts = np.append(tot_dts, [tot_duration])
+                gas_array = np.append(gas_array, [int(gas_cost)])
+            except Exception as e:
+                print(e)
+
+            os.kill(int(p.pid), signal.SIGUSR1)
+
+    after_test = time.time()
+    print("Test total time", after_test - before_test )
+
+    print("-------------- Average: ")
+    print("Dt_gen_proof: "+str(np.mean(proof_dts)))
+    print("Dt_mix: " + str(np.mean(mix_dts)))
+    print("Dt_tx: " + str(np.mean(tx_dts)))
+    print("Dt_tot: " + str(np.mean(tot_dts)))
+    print("gas_cost:" + str(np.mean(gas_array)))
+
+    print("\n-------------- Variance: ")
+    print("Var: ", "Dt_gen_proof: " + str(np.var(proof_dts)))
+    print("Dt_mix: " + str(np.var(mix_dts)))
+    print("Dt_tx: " + str(np.var(tx_dts)))
+    print("Dt_tot: " + str(np.var(tot_dts)))
+    print("gas_cost: " + str(np.var(gas_array)))
```

Comment:
Is it full working? I remember you got problem with the gas consumption. Btw dunno if we want add this in a separate merge request since is about benchmarking.

Reply:
it works only for a limited amount of transactions.
Ideally we should do way more, but I cannot get to refill the parties wallets (even reseting ganache does not work).